pkg/chaosdaemon/tasks/tasks_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package tasks
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaoserr"
+)
+
+type FakeConfig struct {
+	i int
+}
+
+func (f *FakeConfig) Add(a Addable) error {
+	A, OK := a.(*FakeConfig)
+	if OK {
+		f.i += A.i
+		return nil
+	}
+	return errors.Wrapf(ErrCanNotAdd, "expect type : *FakeConfig, got : %T", a)
+}
+
+func (f *FakeConfig) Assign(c Injectable) error {
+	C, OK := c.(*FakeChaos)
+	if OK {
+		C.C.i = f.i
+		return nil
+	}
+	return errors.Wrapf(ErrCanNotAssign, "expect type : *FakeChaos, got : %T", c)
+}
+
+func (f *FakeConfig) New(immutableValues interface{}) (Injectable, error) {
+	temp := immutableValues.(*FakeChaos)
+	f.Assign(temp)
+	return temp, nil
+}
+
+type FakeChaos struct {
+	C              FakeConfig
+	ErrWhenRecover bool
+	ErrWhenInject  bool
+	logger         logr.Logger
+}
+
+func (f *FakeChaos) Inject(pid PID) error {
+	if f.ErrWhenInject {
+		return chaoserr.NotImplemented("inject")
+	}
+	return nil
+}
+
+func (f *FakeChaos) Recover(pid PID) error {
+	if f.ErrWhenRecover {
+		return chaoserr.NotImplemented("recover")
+	}
+	return nil
+}
+
+func TestTasks(t *testing.T) {
+	var log logr.Logger
+
+	zapLog, err := zap.NewDevelopment()
+	if err != nil {
+		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
+	}
+	log = zapr.NewLogger(zapLog)
+
+	m := NewTaskManager(log)
+
+	chaos := FakeChaos{
+		ErrWhenRecover: false,
+		ErrWhenInject:  false,
+		logger:         log,
+	}
+	task1 := FakeConfig{i: 1}
```

Comment:
Could we add a test, where the config is added?

Reply:
`FakeConfig{i: 1}` is the config here. I'm sorry what kind of test you want to add?

Reply:
I mean, summed up, where the `FakeConfig.i` becomes 2 or more 🤔

Reply:
OK, my fault, I miss it. I will add it.

Reply:
Added.